diff --git a/src/core/sync/write_ambient.js b/src/core/sync/write_ambient.js
index 1f2188097ade6fe54d680f7d5702ad3df652dc81..1ca6d02f58f116e6e2c72dd544c11a3724961e58 100644
--- a/src/core/sync/write_ambient.js
+++ b/src/core/sync/write_ambient.js
@@ -12,12 +12,17 @@ const descriptions_dir = fileURLToPath(new URL('../../../src/types/synthetic', i
 
 /** @param {string} filename */
 function read_description(filename) {
-	const content = fs.readFileSync(`${descriptions_dir}/${filename}`, 'utf8');
-	return `/**\n${content
-		.trim()
-		.split('\n')
-		.map((line) => ` * ${line}`)
-		.join('\n')}\n */`;
+	const filepath = `${descriptions_dir}/${filename}`;
+	try {
+		const content = fs.readFileSync(filepath, 'utf8');
+		return `/**\n${content
+			.trim()
+			.split('\n')
+			.map((line) => ` * ${line}`)
+			.join('\n')}\n */`;
+	} catch {
+		return '/** */';
+	}
 }
 
 /**
diff --git a/src/runtime/server/page/render.js b/src/runtime/server/page/render.js
index ff11a9cc92294fec9e3e2543e353f543533210ec..99ddc3435d5d18f7862e4a8fa2de47ff275e00ca 100644
--- a/src/runtime/server/page/render.js
+++ b/src/runtime/server/page/render.js
@@ -274,8 +274,9 @@ export async function render_response({
 						inline_styles.set(filename, css);
 						return;
 					}
-
-					inline_styles.set(filename, css(`${assets}/${paths.app_dir}/immutable/assets`, assets));
+					if (typeof css === 'function') {
+						inline_styles.set(filename, css(`${assets}/${paths.app_dir}/immutable/assets`, assets));
+					}
 				});
 			}
 		}
